### PR TITLE
ArgumentRector split follow up details

### PR DESCRIPTION
**Pull request by TomasVotruba**
Follow up to #451 

### Related Changes

- decouple `ArgumentRector` goup tests to own test cases

### Resolves

- https://github.com/rectorphp/rector/pull/451#discussion_r185551618
- https://github.com/rectorphp/rector/issues/398#issuecomment-385353338

### CODE REVIEW

File: src/Configuration/Rector/ArgumentAdderRecipe.php [hunk 1]
```diff
@@ -7,22 +7,22 @@
     /**
      * @var mixed
      */
-    private $replacement;
+    private $defaultValue;
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply src/Configuration/Rector/ArgumentAdderRecipe.php]: For me it's clear from the configuration in `yml`.

[the remark this answers is not in the document]
Comment (mssimi) [reply src/Configuration/Rector/ArgumentAdderRecipe.php]: I mean default_value in config seems hard to understand, maybe rename it for substitution, argument_value, even value seems good, but not default _something.

[the remark this answers is not in the document]
Comment (mssimi) [reply src/Configuration/Rector/ArgumentAdderRecipe.php]: I suggest variable for variable name and default_value for default value

[the remark this answers is not in the document]
Comment (TomasVotruba) [reply src/Configuration/Rector/ArgumentAdderRecipe.php]: Make use of naming in: https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Node/Param.php

[the remark this answers is not in the document]
Comment (TomasVotruba) [reply src/Configuration/Rector/ArgumentAdderRecipe.php]: I'm looking into this and `default_value` is a configuratoin. As is the Recipe a configuration.

Another names seem confusing to me so I'll keep it this way

[the remark this answers is not in the document]
Comment (mssimi) [reply src/Configuration/Rector/ArgumentAdderRecipe.php]: Consider this code:

```php
final class Foo{
 public function foo($position = 0){ // position is argument and 0 is default_value
 }
}

$foo = new $foo;
$this->foo(5); // 5 is Argument
```
Consider having this conf

```yml
services:
    Rector\Rector\Dynamic\ArgumentAdderRector:
        $argumentChangesByMethodAndType:
            -
                class: 'Foo'
                method: 'compile'
                position: 1 // not sure here if it actually starts from zero or not I am trying to replace first argument here
                default_value: 3
```

Code after this conf would look like

```php
final class Foo{
 public function foo($position = 0){
 }
}

$foo = new $foo;
$this->foo(3);
```

We changed argument value from 5 to 3 we have not touched default_value here, coz this is `ArgumentRector` not `ParameterRector`. Code you linked for inspiration is <b>Parameter</b>, why inspiring by parameter class for argument rector, why not by argument class https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Node/Arg.php.

If this explanation is not enought I'm sorry I have no idea how to make it more clear.

[the remark this answers is not in the document]
Comment (mssimi) [reply src/Configuration/Rector/ArgumentAdderRecipe.php]: I suggest video call here.

[the remark this answers is not in the document]
Comment (TomasVotruba) [reply src/Configuration/Rector/ArgumentAdderRecipe.php]: We still need to cover the case, so test case is enough.

**Pull request merged**